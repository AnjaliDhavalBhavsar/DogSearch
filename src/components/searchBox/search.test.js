import React from "react";
import renderer from "react-test-renderer";

import { SearchBox } from ".";

import { render, fireEvent, screen } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

const setup = () => {
  const utils = render(
    <SearchBox handleChange={(value) => console.log(value)} />
  );
  const input = screen.getByLabelText("SearchInput");
  return {
    input,
    ...utils,
  };
};
test("It should  set input value", () => {
  let mockFn = jest.fn();
  SearchBox.handleChange = mockFn();
  const { input } = setup();
  fireEvent.change(input, { target: { value: "dob" } });
  expect(input.value).toBe("dob");
  expect(mockFn).toHaveBeenCalled();
});

test("search snapshot testing", () => {
  const element = renderer
    .create(<SearchBox handleChange={(value) => console.log(value)} />)
    .toJSON();
  expect(element).toMatchSnapshot();
});
