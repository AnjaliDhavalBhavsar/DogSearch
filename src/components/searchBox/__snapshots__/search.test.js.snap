// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search snapshot testing 1`] = `
<div
  className="searchContainer"
>
  <div
    className="searchText"
  >
    Search
  </div>
  <input
    aria-label="SearchInput"
    className="searchInput"
    onChange={[Function]}
    placeholder="Search for dog "
    type="text"
  />
</div>
`;
